spec/models/bigbluebutton_meeting_spec.rb

```diff
@@ -47,4 +47,176 @@
+          }
+          subject { BigbluebuttonMeeting.last }
+          it("sets creator_id") { subject.creator_id.should eq(user.id) }
+          it("sets creator_name") { subject.creator_name.should eq(user.name) }
+        end
+
+        context "and there are user attributes" do
+          let(:user_attrs) {
+            {
+              meetingID: room.meetingid + "-2",
+              name: room.name + "-2",
+              record: false, # important to be false here
+              creator_name: "can override the creator name",
+              creator_id: -10
+            }
+          }
+          before {
+            room.record_meeting = true
+            expect {
+              BigbluebuttonMeeting.create_meeting_record_from_room(room, {}, server, nil, user_attrs)
+            }.to change{ BigbluebuttonMeeting.count }.by(1)
+          }
+          subject { BigbluebuttonMeeting.last }
+          it("sets meetingid") { subject.meetingid.should eql(room.meetingid + '-2') }
+          it("sets name") { subject.name.should eql(room.name + '-2') }
+          it("sets recorded") { subject.recorded.should be(false) }
+          it("sets creator name") { subject.creator_name.should eql("can override the creator name") }
+          it("sets creator id") { subject.creator_id.should eql(-10) }
+        end
+      end
+
+      context "if there were already old meetings associated with the room, finishes them" do
+        let!(:meeting1) { FactoryGirl.create(:bigbluebutton_meeting, room: room, ended: false, running: true) }
+        let!(:meeting2) { FactoryGirl.create(:bigbluebutton_meeting, room: room, ended: false, running: false) }
+
+        before(:each) {
+          BigbluebuttonMeeting.where(room: room, ended: false).count.should be(2)
+          expect {
+            BigbluebuttonMeeting.create_meeting_record_from_room(room, { metadata: metadata }, server, nil, {})
+          }.to change{ BigbluebuttonMeeting.count }.by(1)
+        }
+        it { BigbluebuttonMeeting.where(room: room).count.should be(3) }
+        it { BigbluebuttonMeeting.where(room: room, ended: false).count.should be(1) }
+        it { BigbluebuttonMeeting.where(room: room, ended: true).count.should be(2) }
+        it { BigbluebuttonMeeting.where(room: room, ended: true, running: false).count.should be(2) }
+      end
+    end
+  end
+
+  describe ".create_meeting_record_from_recording" do
+    let(:server) { FactoryGirl.create(:bigbluebutton_server) }
+    let(:room) { FactoryGirl.create(:bigbluebutton_room) }
+    let!(:recording1) { FactoryGirl.create(:bigbluebutton_recording, meeting_id: nil, server_id: server.id, room_id: room.id) }
+    let!(:recording2) { FactoryGirl.create(:bigbluebutton_recording, meeting_id: nil, server_id: server.id, room_id: nil, name: "no_room") }
+    
+    context "when there is a room_id on the recording" do
+      before(:each) {
+        expect {
+          BigbluebuttonMeeting.create_meeting_record_from_recording(recording1)
```

Comment:
This is called multiple times, one for each `it` block. It would be better to call it just one time within a single `it` block with all tests.